contrib/babelfishpg_tsql/src/databasepropertyex.c

```diff
@@ -36,10 +55,10 @@ databasepropertyex(PG_FUNCTION_ARGS)
 
 	if (strcasecmp(property, "Collation") == 0)
 	{
-		const char *server_collation_name = GetConfigOption("babelfishpg_tsql.server_collation_name", false, false);
+		char *database_collation_name = get_collation_name(PG_GETARG_DATUM(0));
 
-		if (server_collation_name)
-			vch = (*common_utility_plugin_ptr->tsql_varchar_input) (server_collation_name, strlen(server_collation_name), -1);
+		if (database_collation_name)
+			vch = (*common_utility_plugin_ptr->tsql_varchar_input) (database_collation_name, strlen(database_collation_name), -1);
```

Comment:
How is this code working? `get_collation_name` expects Oid input but we are passing string datum!

Reply:
oh! then we can reuse it for `set_db_collation_internal`. Please update it accordingly

Reply:
Updated